### ISSUE TITLE
wrong output reading .bmp image with alpha channel 

### ISSUE THREAD

**Issue by GeorgCode** [opencv/opencv#13093]
- OpenCV = 3.3.0
- Operating System / Platform = Win10
- Compiler = VS2017:

reading a bitmap file (*.bmp) with alpha channel information, the loaded image contains wrong data. Instead of the memory 
B1 G1 R1 A1 | B2 G2 R2 A2 | ....
the memory of the loaded image is
B1 G1 R1 | B2 G2 R2 | .... | A1 | A2 ...  --> with invalid A1 A2 ...

The problem occurrs only in case of bmp file format. Using png file format, everything works fine.

##### Steps to reproduce

```		
// initialize data
cv::Mat testImgGray(16, 10, CV_8UC1);
testImgGray.setTo(0);
testImgGray.at<unsigned char>(cv::Point(0, 0)) = 128;
testImgGray.at<unsigned char>(cv::Point(1, 0)) = 255;
testImgGray.at<unsigned char>(cv::Point(2, 0)) = 128;

// convert to gray to BGRA
cv::Mat testImgBGRA;
cv::cvtColor(testImgGray, testImgBGRA, cv::ColorConversionCodes::COLOR_GRAY2BGRA);

// alpha channel will be 0xff on all the bits
cv::Vec4b expected_0_0(128, 128, 128, 255);
cv::Vec4b expected_0_1(255, 255, 255, 255);
cv::Vec4b expected_0_2(128, 128, 128, 255);

auto value_0_0_check_1 = testImgBGRA.at<cv::Vec4b>(cv::Point(0, 0));
auto value_0_1_check_1 = testImgBGRA.at<cv::Vec4b>(cv::Point(1, 0));
auto value_0_2_check_1 = testImgBGRA.at<cv::Vec4b>(cv::Point(2, 0));
_ASSERTE(value_0_0_check_1 == expected_0_0); // succeed
_ASSERTE(value_0_1_check_1 == expected_0_1); // succeed
_ASSERTE(value_0_2_check_1 == expected_0_2); // succeed

std::string fileName("TestImage.bmp");
cv::imwrite(fileName, testImgBGRA);
auto value_0_0_check_2 = testImgBGRA.at<cv::Vec4b>(cv::Point(0, 0));
auto value_0_1_check_2 = testImgBGRA.at<cv::Vec4b>(cv::Point(1, 0));
auto value_0_2_check_2 = testImgBGRA.at<cv::Vec4b>(cv::Point(2, 0));
_ASSERTE(value_0_0_check_2 == expected_0_0); // succeed
_ASSERTE(value_0_1_check_2 == expected_0_1); // succeed
_ASSERTE(value_0_2_check_2 == expected_0_2); // succeed

cv::Mat testImgReloadBGRA = cv::imread(fileName, -1);
auto value_0_0_check_3 = testImgReloadBGRA.at<cv::Vec4b>(cv::Point(0, 0));
auto value_0_1_check_3 = testImgReloadBGRA.at<cv::Vec4b>(cv::Point(1, 0));
auto value_0_2_check_3 = testImgReloadBGRA.at<cv::Vec4b>(cv::Point(2, 0));
_ASSERTE(value_0_0_check_3 == expected_0_0); // succeed, because 2nd pixel is 0xff
_ASSERTE(value_0_1_check_3 == expected_0_1); // will fail
_ASSERTE(value_0_2_check_3 == expected_0_2); // will fail
```

**Comment by alalek** [comment]
What value of `testImgReloadBGRA.type()`?

**Comment by alalek** [comment]
Also try latest OpenCV code from 3.4 branch. OpenCV 3.3.0 has been released >1 year ago

**Comment by GeorgCode** [comment]
All the header values are equal in the both BGRA images:

  | Name | Value | Type
-- | -- | -- | --
◢ | testImgBGRA | {UINT8, 4 x 10 x 16} | cv::Mat
  | [type] | UINT8 |  
  | [channels] | 4 | int
  | [width] | 10 | int
  | [height] | 16 | int
  | ▶ [data] | 0x09054bc0 "€€€ÿÿÿÿÿ€€€ÿ" | unsigned char *
  | [stride] | 40 | unsigned int
  | ▶ [Raw View] | {flags=1124024344 dims=2 rows=16 ...} | cv::Mat
| | |
◢ | testImgReloadBGRA | {UINT8, 4 x 10 x 16} | cv::Mat
  | [type] | UINT8 |  
  | [channels] | 4 | int
  | [width] | 10 | int
  | [height] | 16 | int
  | ▶ [data] | 0x091d4d40 "€€€ÿÿÿ€€€" | unsigned char *
  | [stride] | 40 | unsigned int
  | ▶ [Raw View] | {flags=1124024344 dims=2 rows=16 ...} | cv::Mat

**Comment by GeorgCode** [comment]
@alalek you are right. This is fixed in 3.4.1. Thanks for you help.
